src/simple_model.jl

```diff
@@ -0,0 +1,272 @@
+export SimpleNLPModel, obj, grad, grad!, cons, cons!, jac_coord, jac, jprod,
+       jprod!, jtprod, jtprod!, hess, hprod, hprod!
+
+type SimpleNLPModel <: AbstractNLPModel
+  meta :: NLPModelMeta
+
+  counters :: Counters
+
+  # Functions
+  f :: Function
+  g :: Function
+  g! :: Function
+  H :: Function
+  Hprod :: Function
+  Hprod! :: Function
+  c :: Function
+  c! :: Function
+  J :: Function
+  Jprod :: Function
+  Jprod! :: Function
+  Jtprod :: Function
+  Jtprod! :: Function
+end
+
+function SimpleNLPModel(x0::Vector, obj::Function; y0::Vector = [],
+    lvar::Vector = [], uvar::Vector = [], lcon::Vector = [], ucon::Vector = [],
+    nnzj::Int = 0, nnzh::Int = 0,
+    grad::Function = (args...)->throw(NotImplementedError("grad")),
+    grad!::Function = (args...)->throw(NotImplementedError("grad!")),
+    hess::Function = (args...)->throw(NotImplementedError("hess")),
+    hprod::Function = (args...)->throw(NotImplementedError("hprod")),
+    hprod!::Function = (args...)->throw(NotImplementedError("hprod!")),
+    cons::Function = (args...)->throw(NotImplementedError("cons")),
+    cons!::Function = (args...)->throw(NotImplementedError("cons!")),
+    jac::Function = (args...)->throw(NotImplementedError("jac")),
+    jprod::Function = (args...)->throw(NotImplementedError("jprod")),
+    jprod!::Function = (args...)->throw(NotImplementedError("jprod!")),
+    jtprod::Function = (args...)->throw(NotImplementedError("jtprod")),
+    jtprod!::Function = (args...)->throw(NotImplementedError("jtprod!")))
+
+  nvar = length(x0)
+  length(lvar) == 0 && (lvar = -Inf*ones(nvar))
+  length(uvar) == 0 && (uvar =  Inf*ones(nvar))
+  ncon = maximum([length(lcon); length(ucon); length(y0)])
+  if nnzh == 0
+    try
+      if ncon == 0
+        A = hess(x0)
+      else
+        A = hess(x0, y0)
```

Comment:
True. I think I'm gonna create a fake y0 even if y0 is specified (it could be zeros).